File: artifacts/original/ProjectOne/src/testing/ContactTest.java
/*
 * Jeremy Cameron
 * CS 320
 * ContactTest class which is used for conducting unit tests for the Contact class.
 */

package testing;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import contacts.Contact;

public class ContactTest {
	
	/*
	 * Tests the successful creation of a Contact object with valid parameters.
	 * Verifies a unique contact ID is generated and that parameters are set correctly.
	 */
	@Test
	public void testContactCreationSuccess() {
		Contact contact = new Contact("Jeremy", "Cameron", "1235157298", "1111 Your St");
		assertNotNull(contact.getId());
		assertTrue(contact.getId().length() <= 10);
		assertEquals("Jeremy", contact.getFirstName());
		assertEquals("Cameron", contact.getLastName());
		assertEquals("1235157298", contact.getPhoneNumber());
		assertEquals("1111 Your St", contact.getAddress());
	}
	
	/*
	 * Tests using an invalid first name.
	 * Verifies that an IllegalArgumentException is thrown when the first name is longer than 10 characters.
	 */
	@Test
	public void testFirstNameInvalid() {
		assertThrows(IllegalArgumentException.class, () -> {
			new Contact("JeremyJeremyJeremy", "Cameron", "1235157298", "1111 Your St");
		});
	}
	
	/*
	 * Tests using an invalid last name.
	 * Verifies that an IllegalArgumentException is thrown when the last name is longer than 10 characters.
	 */
	@Test
	public void testLastNameInvalid() {
		assertThrows(IllegalArgumentException.class, () -> {
			new Contact("Jeremy", "CameronCameronCameron", "1235157298", "1111 Your St");
		});
	}
	
	/*
	 * Tests using an invalid phone number.
	 * Verifies than an IllegalArgumentException is thrown when the phone number is not exactly 10 digits or 
	 * doesn't only contain digits.
	 */
	@Test
	public void testPhoneNumberInvalid() {
		assertThrows(IllegalArgumentException.class, () -> {
			new Contact("Jeremy", "Cameron", "123515729", "1111 Your St");
		});
		assertThrows(IllegalArgumentException.class, () -> {
			new Contact("Jeremy", "Cameron", "12351572982", "1111 Your St");
		});
		assertThrows(IllegalArgumentException.class, () -> {
			new Contact("Jeremy", "Cameron", "1235157298C", "1111 Your St");
		});
	}
	
	/*
	 * Tests using an invalid address.
	 * Verifies that an IllegalArgumentException is thrown when the address is longer than 30 characters.
	 */
	@Test
	public void testAddressInvalid() {
		assertThrows(IllegalArgumentException.class, () -> {
			new Contact("Jeremy", "Cameron", "1235157298", "1111 Your St, Apt E, city, state, zip code");
		});
	}
	
	/*
	 * Tests setting the first name.
	 * Verifies that the first name can be updated correctly and that an IllegalArgumentException is thrown 
	 * if input is invalid.
	 */
	@Test
	public void testSetFirstName() {
		Contact contact = new Contact("Jeremy", "Cameron", "1235157298", "1111 Your St");
		contact.setFirstName("Jamie");
		assertEquals("Jamie", contact.getFirstName());
		assertThrows(IllegalArgumentException.class, () -> {
			contact.setFirstName(null);
		});
		assertThrows(IllegalArgumentException.class, () -> {
			contact.setFirstName("JamieCameronJamieCameron");
		});
	}
	
	/*
	 * Tests setting the last name.
	 * Verifies that the last name can be updated correctly and that an IllegalArgumentException is thrown
	 * if input is invalid.
	 */
	@Test
	public void testSetLastName() {
		Contact contact = new Contact("Jeremy", "Cameron", "1235157298", "1111 Your St");
		contact.setLastName("Welker");
		assertEquals("Welker", contact.getLastName());
		assertThrows(IllegalArgumentException.class, () -> {
			contact.setLastName(null);
		});
		assertThrows(IllegalArgumentException.class, () -> {
			contact.setLastName("WelkerWelkerWelker");
		});
	}
	
	/*
	 * Tests setting the phone number.
	 * Verifies that the phone number can be updated correctly and that an IllegalArgumentException is thrown
	 * if input is invalid.
	 */
	@Test
	public void testSetPhoneNumber() {
		Contact contact = new Contact("Jeremy", "Cameron", "1235157298", "1111 Your St");
		contact.setPhoneNumber("2226117828");
		assertEquals("2226117828", contact.getPhoneNumber());
		assertThrows(IllegalArgumentException.class, () -> {
			contact.setPhoneNumber(null);
		});
		assertThrows(IllegalArgumentException.class, () -> {
			contact.setPhoneNumber("123515729");
		});
		assertThrows(IllegalArgumentException.class, () -> {
			contact.setPhoneNumber("22261178288");
		});
		assertThrows(IllegalArgumentException.class, () -> {
			contact.setPhoneNumber("kekuwurofl");
		});
	}
	
	/*
	 * Tests setting the address.
	 * Verifies that the address can be updated correctly and that an IllegalArgumentException is thrown
	 * if input is invalid.
	 */
	@Test
	public void testSetAddress() {
		Contact contact = new Contact("Jeremy", "Cameron", "1235157298", "1111 Your St");
		contact.setAddress("2222 My Ave");
		assertEquals("2222 My Ave", contact.getAddress());
		assertThrows(IllegalArgumentException.class, () -> {
			contact.setAddress(null);
		});
		assertThrows(IllegalArgumentException.class, () -> {
			contact.setAddress("1111 My Ave, Apt G, city, state, zip code");
		});
	}
}

File: enhancement_1/CS499_Project/src/Tests/contact.test.js
// tests/contact.test.js
import Contact from '../domain/contact.js';

describe('Contact', () => {

  // Test for successfully creating a new contact
  test('creates contact with valid parameters', () => {
    const contact = new Contact({
      firstName: 'Jeremy',
      lastName: 'Cameron',
      phoneNumber: '1235157298',
      address: '1111 Your St'
    });

    expect(contact.id).toBeDefined();
    expect(contact.id.length).toBeLessThanOrEqual(10);
    expect(contact.firstName).toBe('Jeremy');
    expect(contact.lastName).toBe('Cameron');
    expect(contact.phoneNumber).toBe('1235157298');
    expect(contact.address).toBe('1111 Your St');
  });

  // Test for throwing an error when the first name is too long
  test('throws error for invalid first name (too long)', () => {
    expect(() => {
      new Contact({
        firstName: 'JeremyJeremyJeremy',
        lastName: 'Cameron',
        phoneNumber: '1235157298',
        address: '1111 Your St'
      });
    }).toThrow();
  });

  // Test for throwing an error when the last name is too long
  test('throws error for invalid last name (too long)', () => {
    expect(() => {
      new Contact({
        firstName: 'Jeremy',
        lastName: 'CameronCameronCameron',
        phoneNumber: '1235157298',
        address: '1111 Your St'
      });
    }).toThrow();
  });

  // Test for throwing an error when the phone number is invalid
  test('throws error for invalid phone number (not exactly 10 digits or contains letters)', () => {
    expect(() => {
      new Contact({
        firstName: 'Jeremy',
        lastName: 'Cameron',
        phoneNumber: '123515729',
        address: '1111 Your St'
      });
    }).toThrow();

    expect(() => {
      new Contact({
        firstName: 'Jeremy',
        lastName: 'Cameron',
        phoneNumber: '12351572982',
        address: '1111 Your St'
      });
    }).toThrow();

    expect(() => {
      new Contact({
        firstName: 'Jeremy',
        lastName: 'Cameron',
        phoneNumber: '1235157298C',
        address: '1111 Your St'
      });
    }).toThrow();
  });

  // Test for throwing an error when the address is too long
  test('throws error for invalid address (too long)', () => {
    expect(() => {
      new Contact({
        firstName: 'Jeremy',
        lastName: 'Cameron',
        phoneNumber: '1235157298',
        address: '1111 Your St, Apt E, city, state, zip code'
      });
    }).toThrow();
  });

  // Test for successfully updating a contact's first name and throwing an error if the input is too long
  test('setFirstName updates name and throws on invalid input', () => {
    const contact = new Contact({
      firstName: 'Jeremy',
      lastName: 'Cameron',
      phoneNumber: '1235157298',
      address: '1111 Your St'
    });
    contact.setFirstName('Jamie');
    expect(contact.firstName).toBe('Jamie');

    expect(() => contact.setFirstName(null)).toThrow();
    expect(() => contact.setFirstName('JamieCameronJamieCameron')).toThrow();
  });

  // Test for successfully updating a contact's last name and throwing an error if the input is too long
  test('setLastName updates name and throws on invalid input', () => {
    const contact = new Contact({
      firstName: 'Jeremy',
      lastName: 'Cameron',
      phoneNumber: '1235157298',
      address: '1111 Your St'
    });
    contact.setLastName('Welker');
    expect(contact.lastName).toBe('Welker');

    expect(() => contact.setLastName(null)).toThrow();
    expect(() => contact.setLastName('WelkerWelkerWelker')).toThrow();
  });

  // Test for successfully updating a contact's phone number and throwing an error if the input is too long or contains letters
  test('setPhoneNumber updates number and throws on invalid input', () => {
    const contact = new Contact({
      firstName: 'Jeremy',
      lastName: 'Cameron',
      phoneNumber: '1235157298',
      address: '1111 Your St'
    });
    contact.setPhoneNumber('2226117828');
    expect(contact.phoneNumber).toBe('2226117828');

    expect(() => contact.setPhoneNumber(null)).toThrow();
    expect(() => contact.setPhoneNumber('123515729')).toThrow();
    expect(() => contact.setPhoneNumber('22261178288')).toThrow();
    expect(() => contact.setPhoneNumber('kekuwurofl')).toThrow();
  });

  // Test for successfully updating a contact's address and throwing an error if the input is too long
  test('setAddress updates address and throws on invalid input', () => {
    const contact = new Contact({
      firstName: 'Jeremy',
      lastName: 'Cameron',
      phoneNumber: '1235157298',
      address: '1111 Your St'
    });
    contact.setAddress('2222 My Ave');
    expect(contact.address).toBe('2222 My Ave');

    expect(() => contact.setAddress(null)).toThrow();
    expect(() => contact.setAddress('1111 My Ave, Apt G, city, state, zip code')).toThrow();
  });
});
